src/target/target_flash.c

```diff
@@ -174,6 +185,65 @@ bool target_flash_erase(target_s *target, target_addr_t addr, size_t len)
 	return result;
 }
 
+static inline bool flash_manual_mass_erase(target_flash_s *const flash, platform_timeout_s *const print_progess)
+{
+	for (target_addr_t addr = flash->start; addr < (flash->start + flash->length); addr += flash->blocksize) {
+		if (!flash->erase(flash, addr, flash->blocksize))
+			return false;
+		target_print_progress(print_progess);
+	}
+	return true;
+}
+
+/* Run specialized target mass erase if available, otherwise erase all flash' */
+bool target_flash_mass_erase(target_s *const target)
+{
+	if (!target_enter_flash_mode(target))
+		return false;
+
+	/* Setup progress printout */
+	platform_timeout_s print_progess;
+	platform_timeout_set(&print_progess, 500U);
+
+	bool result = false;
+	if (target->mass_erase) {
+		DEBUG_INFO("Running specialized target mass erase\n");
```

Comment:
Given this appears to be for target debugging, perhaps let's drop it to the `DEBUG_TARGET()` log level? A user running BMD under `-v 1` to get diagnostics on information like the ROM tables for their part and general information possibly don't want diagnostics about how their part is to be mass erased, that feels more like if a user wants diagnostics on how BMD is driving their target.